contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -1005,16 +1019,23 @@ contract RibbonTreasuryVault is
     function _distributePremium(IERC20 token, uint256 amount) internal {
         // Distribute to whitelisted address
         address[] storage _whitelist = whitelistArray;
+        uint256[] memory _amounts = new uint256[](_whitelist.length);
         uint256 totalSupply = totalSupply();
 
         for (uint256 i = 0; i < _whitelist.length; i++) {
             // Distribute to whitelist proportional to the amount of
             // shares they own
-            token.safeTransfer(
-                _whitelist[i],
-                shares(_whitelist[i]).mul(amount).div(totalSupply)
-            );
+            _amounts[i] = shares(_whitelist[i]).mul(amount).div(totalSupply);
+
+            token.safeTransfer(_whitelist[i], _amounts[i]);
```

Comment:
Accessing _whitelist[i] twice here so we can cache it in a variable. Alternatively we can keep to how we were writing before

Reply:
I'm caching it in a variable, reason is to store the distributed amount into the array for easier subgraph mapping.